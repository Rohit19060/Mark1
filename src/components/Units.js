import React, { Component } from "react";
import Unit from "./Unit";

class Units extends Component {
  render() {
    return this.props.units.map((unit) => (
      <Unit unit={unit} key={unit.id} deleteUnit={this.props.deleteUnit} />
    ));
  }
}

export default Units;
