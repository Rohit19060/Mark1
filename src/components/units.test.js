import Units from "./Units";
import React from "react";
import { BrowserRouter as Router } from "react-router-dom";
import { render } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import fs from "fs";

test("Check Units", () => {
  let rawdata = fs.readFileSync("server/units.json");
  let units = JSON.parse(rawdata);
  let ourUnits = units.units;
  const deleteUnit = jest.fn();
  const component = render(
    <Router>
      <Units units={ourUnits} deleteUnit={deleteUnit} />
    </Router>
  );
  ourUnits.map((c) => expect(component.container).toHaveTextContent(c.title));
});

test("snapshot test", () => {
  let rawdata = fs.readFileSync("server/units.json");
  let units = JSON.parse(rawdata);
  let ourUnits = units.units;
  const deleteUnit = jest.fn();
  const component = render(
    <Router>
      <Units units={ourUnits} deleteUnit={deleteUnit} />
    </Router>
  );
  expect(component).toMatchSnapshot();
});
