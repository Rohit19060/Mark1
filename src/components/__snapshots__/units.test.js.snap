// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="row"
      >
        <h5
          class="six columns"
          data-testid="COMP1750"
        >
          <a
            href="/unit/COMP1750"
          >
            COMP1750
          </a>
          - 
          Introduction to Business Information Systems
        </h5>
        <button
          class="button-primary two columns"
        >
          Up
        </button>
         
        <button
          class="button-primary two columns"
        >
          Down
        </button>
         
        <button
          class="button-primary two columns"
        >
          Delete
        </button>
      </div>
      <div
        class="row"
      >
        <h5
          class="six columns"
          data-testid="COMP2110"
        >
          <a
            href="/unit/COMP2110"
          >
            COMP2110
          </a>
          - 
          Web Technology
        </h5>
        <button
          class="button-primary two columns"
        >
          Up
        </button>
         
        <button
          class="button-primary two columns"
        >
          Down
        </button>
         
        <button
          class="button-primary two columns"
        >
          Delete
        </button>
      </div>
      <div
        class="row"
      >
        <h5
          class="six columns"
          data-testid="COMP2750"
        >
          <a
            href="/unit/COMP2750"
          >
            COMP2750
          </a>
          - 
          Applications Modelling and Development
        </h5>
        <button
          class="button-primary two columns"
        >
          Up
        </button>
         
        <button
          class="button-primary two columns"
        >
          Down
        </button>
         
        <button
          class="button-primary two columns"
        >
          Delete
        </button>
      </div>
      <div
        class="row"
      >
        <h5
          class="six columns"
          data-testid="MMCC2045"
        >
          <a
            href="/unit/MMCC2045"
          >
            MMCC2045
          </a>
          - 
          Interactive Web Design
        </h5>
        <button
          class="button-primary two columns"
        >
          Up
        </button>
         
        <button
          class="button-primary two columns"
        >
          Down
        </button>
         
        <button
          class="button-primary two columns"
        >
          Delete
        </button>
      </div>
      <div
        class="row"
      >
        <h5
          class="six columns"
          data-testid="COMP3120"
        >
          <a
            href="/unit/COMP3120"
          >
            COMP3120
          </a>
          - 
          Advanced Web Development
        </h5>
        <button
          class="button-primary two columns"
        >
          Up
        </button>
         
        <button
          class="button-primary two columns"
        >
          Down
        </button>
         
        <button
          class="button-primary two columns"
        >
          Delete
        </button>
      </div>
      <div
        class="row"
      >
        <h5
          class="six columns"
          data-testid="asdf"
        >
          <a
            href="/unit/asdf"
          >
            asdf
          </a>
          - 
          asdf
        </h5>
        <button
          class="button-primary two columns"
        >
          Up
        </button>
         
        <button
          class="button-primary two columns"
        >
          Down
        </button>
         
        <button
          class="button-primary two columns"
        >
          Delete
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="row"
    >
      <h5
        class="six columns"
        data-testid="COMP1750"
      >
        <a
          href="/unit/COMP1750"
        >
          COMP1750
        </a>
        - 
        Introduction to Business Information Systems
      </h5>
      <button
        class="button-primary two columns"
      >
        Up
      </button>
       
      <button
        class="button-primary two columns"
      >
        Down
      </button>
       
      <button
        class="button-primary two columns"
      >
        Delete
      </button>
    </div>
    <div
      class="row"
    >
      <h5
        class="six columns"
        data-testid="COMP2110"
      >
        <a
          href="/unit/COMP2110"
        >
          COMP2110
        </a>
        - 
        Web Technology
      </h5>
      <button
        class="button-primary two columns"
      >
        Up
      </button>
       
      <button
        class="button-primary two columns"
      >
        Down
      </button>
       
      <button
        class="button-primary two columns"
      >
        Delete
      </button>
    </div>
    <div
      class="row"
    >
      <h5
        class="six columns"
        data-testid="COMP2750"
      >
        <a
          href="/unit/COMP2750"
        >
          COMP2750
        </a>
        - 
        Applications Modelling and Development
      </h5>
      <button
        class="button-primary two columns"
      >
        Up
      </button>
       
      <button
        class="button-primary two columns"
      >
        Down
      </button>
       
      <button
        class="button-primary two columns"
      >
        Delete
      </button>
    </div>
    <div
      class="row"
    >
      <h5
        class="six columns"
        data-testid="MMCC2045"
      >
        <a
          href="/unit/MMCC2045"
        >
          MMCC2045
        </a>
        - 
        Interactive Web Design
      </h5>
      <button
        class="button-primary two columns"
      >
        Up
      </button>
       
      <button
        class="button-primary two columns"
      >
        Down
      </button>
       
      <button
        class="button-primary two columns"
      >
        Delete
      </button>
    </div>
    <div
      class="row"
    >
      <h5
        class="six columns"
        data-testid="COMP3120"
      >
        <a
          href="/unit/COMP3120"
        >
          COMP3120
        </a>
        - 
        Advanced Web Development
      </h5>
      <button
        class="button-primary two columns"
      >
        Up
      </button>
       
      <button
        class="button-primary two columns"
      >
        Down
      </button>
       
      <button
        class="button-primary two columns"
      >
        Delete
      </button>
    </div>
    <div
      class="row"
    >
      <h5
        class="six columns"
        data-testid="asdf"
      >
        <a
          href="/unit/asdf"
        >
          asdf
        </a>
        - 
        asdf
      </h5>
      <button
        class="button-primary two columns"
      >
        Up
      </button>
       
      <button
        class="button-primary two columns"
      >
        Down
      </button>
       
      <button
        class="button-primary two columns"
      >
        Delete
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
